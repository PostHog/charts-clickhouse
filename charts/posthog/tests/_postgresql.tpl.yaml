# Tests charts/posthog/templates/_postgresql.tpl
suite: _postgresql.tpl template
templates:
  - templates/test-templates.yaml

tests:
  - it: should render with defaults
    set:
      testTemplates.postgresqlEnvTest: true
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot:
          path: data

  - it: should render with custom values
    set:
      testTemplates.postgresqlEnvTest: true
      postgresql.nameOverride: customName
      postgresql.postgresqlDatabase: customDb
      postgresql.postgresqlUsername: customUser
      postgresql.postgresqlPassword: customPassword
    asserts:
      - matchSnapshot:
          path: data

  - it: should render with existing secret
    set:
      testTemplates.postgresqlEnvTest: true
      postgresql.existingSecret: someExistingSecret
    asserts:
      - matchSnapshot:
          path: data

  - it: should render with external postgresql and password
    set:
      testTemplates.postgresqlEnvTest: true
      postgresql.enabled: false
      externalPostgresql.postgresqlHost: pg.foo.bar
      externalPostgresql.postgresqlDatabase: somedb
      externalPostgresql.postgresqlUsername: someuser
      externalPostgresql.postgresqlPassword: password123
    asserts:
      - matchSnapshot:
          path: data

  - it: should render with external postgresql and existing secret
    set:
      testTemplates.postgresqlEnvTest: true
      postgresql.enabled: false
      externalPostgresql.postgresqlHost: pg.foo.bar
      externalPostgresql.postgresqlDatabase: somedb
      externalPostgresql.postgresqlUsername: someuser
      externalPostgresql.existingSecret: password123
      externalPostgresql.existingSecretPasswordKey: somekey
    asserts:
      - matchSnapshot:
          path: data

  - it: should raise when postgresql is disabled but no external postgresql
    set:
      testTemplates.postgresqlEnvTest: true
      postgresql.enabled: false
    asserts:
      - failedTemplate:
          errorMessage: externalPostgresql.postgresqlHost is required if not postgresql.enabled
