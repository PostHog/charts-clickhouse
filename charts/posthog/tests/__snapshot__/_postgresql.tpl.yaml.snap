should render with custom values:
  1: |
    - name: POSTHOG_DB_USER
      value: customUser
    - name: POSTHOG_DB_NAME
      value: customDb
    - name: POSTHOG_DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: postgresql-password
          name: RELEASE-NAME-customName
    - name: POSTHOG_POSTGRES_HOST
      value: RELEASE-NAME-customName
    - name: POSTHOG_POSTGRES_PORT
      value: "5432"
    - name: USING_PGBOUNCER
      value: "false"
should render with defaults:
  1: |
    - name: POSTHOG_DB_USER
      value: postgres
    - name: POSTHOG_DB_NAME
      value: posthog
    - name: POSTHOG_DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: postgresql-password
          name: RELEASE-NAME-posthog-postgresql
    - name: POSTHOG_POSTGRES_HOST
      value: RELEASE-NAME-posthog-postgresql
    - name: POSTHOG_POSTGRES_PORT
      value: "5432"
    - name: USING_PGBOUNCER
      value: "false"
should render with existing secret:
  1: |
    - name: POSTHOG_DB_USER
      value: postgres
    - name: POSTHOG_DB_NAME
      value: posthog
    - name: POSTHOG_DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: clickhousePassword
          name: someExistingSecret
    - name: POSTHOG_POSTGRES_HOST
      value: RELEASE-NAME-posthog-postgresql
    - name: POSTHOG_POSTGRES_PORT
      value: "5432"
    - name: USING_PGBOUNCER
      value: "false"
